lib/elixir/lib/macro.ex

```diff
@@ -843,6 +843,66 @@ defmodule Macro do
   `escape/2` is used to escape *values* (either directly passed or variable
   bound), while `quote/2` produces syntax trees for
   expressions.
+
+  ## Dealing with references and other runtime values
+
+  Macros work at compile-time and therefore `Macro.escape/1` can only escape values
+  that are valid during compilation, such as numbers, atoms, tuples, maps, binaries,
+  etc.
+
+  However, you may have values at compile-time which cannot be escaped, such as
+  `reference`s and `pid`s, since the process or memory address they point to will
+  no longer exist once compilation completes. Attempting to escape said values will
+  raise. This is a common issue when working with NIFs.
```

Comment:
```suggestion
  lead to errors. This is a common issue when working with NIFs.
```

Reply:
This is probably too vague. Let's go with
```suggestion
  raise an exception. This is a common issue when working with NIFs.
```